kotlinx-coroutines-core/jvm/src/scheduling/CoroutineScheduler.kt

```diff
@@ -730,23 +705,17 @@ internal class CoroutineScheduler(
         // Counterpart to "tryUnpark"
         private fun tryPark() {
             if (!inStack()) {
-                parkingState.value = PARKING_ALLOWED
                 parkedWorkersStackPush(this)
                 return
-
             }
             assert { localQueue.size == 0 }
-            // Failed to get a parking permit => we are not in the stack
             while (inStack()) {
```

Comment:
Performance nit: Before this line we've already checked that `inStack()` is true, so this loop can be replaced with 
`do { ... } while (inStack())`